keyboards/trojan_pinata/model_b/rev0/rev0.c

```diff
@@ -0,0 +1,589 @@
+/* Copyright 2024 Brian H (@Trojan_Pinata)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "rev0.h"
+
+enum oled_modes {
+    MTNS, INFO, LOGO, MEW, CEL, RAY
+};
+
+#define NUM_MODES 6
+uint8_t display_mode = INFO;    // defaults to 1 in case of no value
+
+// key that changes OLED
+#define OLED_MOD KC_F23
+
+// virtual eeprom for storing last layer
+typedef union {
+    uint32_t raw;
+    struct {
+        uint8_t display_mode_saved :NUM_MODES;
```

Comment:
This is wrong. This allocates 6 bits for the variable, which is 64 possible values.  You want the highest number of bits that support it, which would be 3, in this case (this supports 8 values, 0-7). 4 would give you 16 values. 
```suggestion
        uint8_t display_mode_saved : 3;
```